core/src/main/java/org/mule/runtime/core/privileged/exception/TemplateOnErrorHandler.java

```diff
@@ -280,10 +288,12 @@ protected CoreEvent nullifyExceptionPayloadIfRequired(CoreEvent event) {
   @Override
   protected void doInitialise() throws InitialisationException {
     super.doInitialise();
-    Optional<ProcessingStrategy> processingStrategy = empty();
-    if (flowLocation.isPresent()) {
+    Optional<ProcessingStrategy> processingStrategy;
+    if (fromGlobalErrorHandler && parseBoolean(getProperty(REUSE_GLOBAL_ERROR_HANDLER_PROPERTY))) {
+      processingStrategy = getProcessingStrategyFromGlobalErrorHandler(locator);
```

Comment:
this should be eventually replaced by the proper PS propagation, right? Is there any task for this to lint it with a TODO?

cc @fsgonz